pkg/auth/user/UserService.go

```diff
@@ -944,15 +949,103 @@ func (impl *UserServiceImpl) GetAll() ([]bean.UserInfo, error) {
 	return response, nil
 }
 
-func (impl *UserServiceImpl) GetAllDetailedUsers() ([]bean.UserInfo, error) {
-	models, err := impl.userRepository.GetAllExcludingApiTokenUser()
+// GetAllWithFilters takes filter arguments gives UserListingResponse as output with some operations like filter, sorting, searching,pagination support inbuilt
+func (impl UserServiceImpl) GetAllWithFilters(sortOrder string, sortBy string, offset int, totalSize int, showAll bool, searchKey string) (*bean.UserListingResponse, error) {
+	//  default values will be used if not provided
+	request := impl.getRequestWithFiltersArgsOrDefault(sortOrder, sortBy, offset, totalSize, showAll, searchKey)
+	if request.ShowAll {
+		return impl.getAllDetailedUsers()
+	}
+	// Setting size as zero to calculate the total number of results based on request
+	size := request.Size
+	request.Size = 0
+
+	// Build query from query builder
+	query := impl.userListingRepositoryQueryBuilder.GetQueryForUserListingWithFilters(request)
+	totalCount, err := impl.userRepository.GetCountExecutingQuery(query)
 	if err != nil {
 		impl.logger.Errorw("error while fetching user from db", "error", err)
 		return nil, err
 	}
+	request.Size = size
+
+	query = impl.userListingRepositoryQueryBuilder.GetQueryForUserListingWithFilters(request)
+	models, err := impl.userRepository.GetAllExecutingQuery(query)
+	if err != nil {
+		impl.logger.Errorw("error while fetching user from db", "error", err)
+		return nil, err
+	}
+
+	response, err := impl.getUserResponse(models)
+	if err != nil {
+		impl.logger.Errorw("error in getUserResponseWithLoginAudit", "err", err)
+		return nil, err
+	}
+
+	listingResponse := &bean.UserListingResponse{
+		Users:      response,
+		TotalCount: totalCount,
+	}
+	return listingResponse, nil
+
+}
+
+func (impl UserServiceImpl) getRequestWithFiltersArgsOrDefault(sortOrder string, sortBy string, offset int, totalSize int, showAll bool, searchKey string) *bean.FetchListingRequest {
+	sortByRes, size := impl.userCommonService.GetDefaultValuesIfNotPresent(sortBy, totalSize, false)
+	request := &bean.FetchListingRequest{
+		SortOrder: bean2.SortOrder(sortOrder),
+		SortBy:    sortByRes,
+		Offset:    offset,
+		Size:      size,
+		ShowAll:   showAll,
+		SearchKey: searchKey,
+	}
+	return request
+}
+
+func (impl UserServiceImpl) getAllDetailedUsers() (*bean.UserListingResponse, error) {
```

Comment:
just working as an adapter, should be independent of (impl UserServiceImpl)

Reply:
DONE